external/radiation/radiation/wrapper_api.py

```diff
@@ -247,13 +247,13 @@ def _rad_compute(self, state: State, time: cftime.DatetimeJulian,) -> Diagnostic
             self._cached = unstack(postprocess_out(out), coords)
         return self._cached
 
-
-def _solar_hour(time: cftime.DatetimeJulian) -> float:
-    return (
-        time.hour
-        + time.minute / MINUTES_PER_HOUR
-        + time.second / (MINUTES_PER_HOUR * SECONDS_PER_MINUTE)
-    )
+    def _solar_hour(self, time: cftime.DatetimeJulian) -> float:
+        """This follows the Fortran computation that rounds initial time to the
+        hour. See https://github.com/NOAA-GFDL/SHiELD_physics/issues/17, but for
+        now we want the port to validate against Fortran."""
```

Comment:
I don't see any rounding of the time to the nearest hour in this function.

Reply:
Here is the line:
```
Model%solhr  = mod(Model%phour+Model%idate(1),con_24)
```
This takes the fractional hours since initialization (here "initialization" includes hours+minutes+seconds), and adds it to *only the hour* (`idate(1)`) of the initialization, which effectively subtracts any minutes+seconds in the initialization time from the resulting solar hour.

Reply:
I see, thanks for explanation. I think either description works—although maybe a unit test would be good to ensure the behavior stays as expected.